modules/python/python-http-header.c

```diff
@@ -0,0 +1,383 @@
+/*
+ * Copyright (c) 2020 One Identity
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "python-http-header.h"
+#include "python-helpers.h"
+
+#include "driver.h"
+#include "str-utils.h"
+#include "list-adt.h"
+
+#include "modules/http/http-signals.h"
+
+#include <time.h>
+
+#define PYTHON_HTTP_HEADER_PLUGIN "python-http-header"
+
+struct _PythonHttpHeaderPlugin
+{
+  LogDriverPlugin super;
+
+  gchar *class;
+  GList *loaders;
+
+  GList *last_headers;
+  time_t last_run;
+  int timeout;
+
+  GHashTable *options;
+
+  struct
+  {
+    PyObject *class;
+    PyObject *instance;
+    PyObject *get_headers;
+  } py;
+};
+
+static gboolean
+_py_append_pylist_to_list(PyObject *py_list, GList **list)
+{
+  const gchar *str;
+  PyObject *py_str;
+
+  if (!py_list || !PyList_Check(py_list))
+    goto exit;
+
+  Py_ssize_t len = PyList_Size(py_list);
+  for (int i = 0; i < len; i++)
+    {
+      py_str = PyList_GetItem(py_list, i); // Borrowed reference
+      if (!py_str || !PyUnicode_Check(py_str))
```

Comment:
This does not work with python2. There is a `_py_is_string` function that could be used.